### PR TITLE
Update DG images

### LINKED ISSUE #74: Standardise the DG diagrams
In general, each DG diagram should begin with a description, followed by the diagram.

In the description, it should reference the diagram as well. ~~Of course, each diagram has to be labelled.~~

EDIT: I think labelling the diagram may need too much work, as we need to standardise within the whole DG. It is also quite annoying as markdown doesn't really support a nice image caption below the image. We can do it for the `.pdf` version?

### PR DESCRIPTION
**Pull request by jieffyy**
Standardise and update the DG images. Closes #74.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -51,22 +58,24 @@ For example, the `Logic` component (see the class diagram given below) defines i
 
 **How the architecture components interact with each other**
```
Comment (yanchenglee98) [inline docs/DeveloperGuide.md:59]: This statement should reference the picture above. Currently the picture `LogicClassDiagram.png` is not referenced in the DG

**Review by yanchenglee98 — COMMENTED** [review]
Apart from a missing sequence diagram reference, LGTM.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (jieffyy) [reply docs/DeveloperGuide.md:59]: Okay done. Thanks!

**Review by yanchenglee98 — APPROVED** [review]
LGTM

**Pull request merged**